src/Control/Carrier/Interpret.hs

```diff
@@ -36,9 +35,9 @@ import           Control.Monad.Trans.Class
 import           Data.Functor.Const (Const(..))
 import           Unsafe.Coerce (unsafeCoerce)
 
--- | A @Handler@ is a function that interprets effects described by @sig@ into the carrier monad @m@.
-newtype Handler sig m = Handler
-  { runHandler :: forall s n x . Monad n => (forall y . n y -> InterpretC s sig m y) -> sig n x -> InterpretC s sig m x }
+-- | An @Interpreter@ is a function that interprets effects described by @sig@ into the carrier monad @m@.
+newtype Interpreter sig m = Interpreter
+  { runInterpreter :: forall ctx n s x . Functor ctx => Handler ctx n (InterpretC s sig m) -> sig n x -> ctx () -> InterpretC s sig m (ctx x) }
```

Comment:
This was never a handler (which is what we call functions like `runState`), but rather an interpreter which now _takes_ a handler.